src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.T.cs

```diff
@@ -2148,14 +2147,41 @@ public static void Round<T>(ReadOnlySpan<T> x, MidpointRounding mode, Span<T> de
         public static void Round<T>(ReadOnlySpan<T> x, int digits, MidpointRounding mode, Span<T> destination)
             where T : IFloatingPoint<T>
         {
+            if (digits == 0)
+            {
+                switch (mode)
+                {
+                    case MidpointRounding.ToZero:
+                        Truncate(x, destination);
+                        return;
+
+                    case MidpointRounding.ToNegativeInfinity:
+                        Floor(x, destination);
+                        return;
+
+                    case MidpointRounding.ToPositiveInfinity:
+                        Ceiling(x, destination);
+                        return;
+
+                    case MidpointRounding.AwayFromZero:
+                    case MidpointRounding.ToEven:
+                        // TODO: Vectorize the remaining modes
+                        break;
+                }
+            }
+
             if (x.Length > destination.Length)
             {
                 ThrowHelper.ThrowArgument_DestinationTooShort();
             }
 
+            if ((uint)mode > (uint)MidpointRounding.ToPositiveInfinity)
+            {
+                double.Round(0, 0, mode); // throw same exception for invalid enum as built-in types do
```

Comment:
It's probably better to duplicate the logic so its easier for the JIT to see that this path definitively throws and can mark the path as `cold`.